pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -760,3 +961,85 @@ func checkNodeAffinity(og *operationGenerator, volumeToMount VolumeToMount, plug
 	}
 	return nil
 }
+
+func mapDevice(devicePath string, globalPDPath string, mounter mount.Interface) error {
+	// globalPDPath
+	// ex. /var/lib/kubelet/plugins/kubernetes.io/<plugin>/<volume>
+	if len(devicePath) == 0 {
+		return fmt.Errorf("Failed to map device to global mount path. devicePath is empty.")
+	}
+	if len(globalPDPath) == 0 {
+		return fmt.Errorf("Failed to map device to global mount path. globalPDPath is empty.")
+	}
+
+	glog.Errorf("#### DEBUG LOG ####: mapDevice devicePath: %s", devicePath)
+	glog.Errorf("#### DEBUG LOG ####: mapDevice globalPDPath; %s", globalPDPath)
+
+	noMnt, err := mounter.IsLikelyNotMountPoint(globalPDPath)
+	if !noMnt {
+		return fmt.Errorf("%s already mounted. This volume cant' be used as raw block volume.", globalPDPath)
+	}
+	if err != nil && !os.IsNotExist(err) {
+		glog.Errorf("cannot validate global mount path: %s", globalPDPath)
+		return err
+	}
+	if err = os.MkdirAll(globalPDPath, 0750); err != nil {
+		return fmt.Errorf("Failed to mkdir %s, error", globalPDPath)
+	}
+	if err := os.Symlink(devicePath, globalPDPath+"/"+"symlink"); err != nil && !os.IsExist(err) {
+		return err
+	}
+	glog.Errorf("#### DEBUG LOG ####: mapDevice: no error")
+	return nil
+}
+
+func unmapDevice(globalPDPath string) error {
```

Comment:
Shouldn't the plugin be doing these?

Reply:
If we implement this method here, each plugin doesn't need to implement this method individually.
But since this method has OS dependent implementation like os.lstat, os.remove, this might better to be implemented at individual plugin.
Any idea?

Reply:
If we want to follow the same model as filesystem volumes, then the plugin should do all the work.  But I do see value in having kubelet handle common methods that are used by all plugins.  At this point, do we know for sure that there are common steps that are the same for all plugins?

Reply:
@msau42 
The clean up steps are under discussion with @jsafrane on the spec doc.
Could you give feedback?

https://github.com/kubernetes/community/pull/805/files#diff-b828f85c1086b7acfaf489fe8605329fR609